contracts/Org.sol

```diff
@@ -54,40 +58,67 @@ contract Org is Administratable {
 
   /**
    * @notice Create Organization Claim
+   * @param  claimId UUID representing this claim
    * @param  fName First name of Administrator
    * @param  lName Last name of Administrator
    * @param  eMail Email contact for Organization Administrator.
    * @param  orgAdminWalletAddress Wallet address of Organization's Administrator.
    */
   function claimRequest(
+    string memory claimId,
     string memory fName,
     string memory lName,
     string memory eMail,
     address orgAdminWalletAddress
   ) public {
+    require(
+      pendingClaims[claimId].desiredWallet == address(0),
+      "Org: Pending Claim with Id already exists"
+    );
+
     Claim memory newClaim = Claim({
       firstName: fName,
       lastName: lName,
       eMail: eMail,
-      desiredWallet: orgAdminWalletAddress,
-      filesSubmitted: true
+      desiredWallet: orgAdminWalletAddress
     });
-    emit ClaimCreated(newClaim);
-    claims.push(newClaim);
+
+    emit ClaimCreated(claimId, newClaim);
+    pendingClaims[claimId] = newClaim;
   }
 
   /**
    * @notice Approving Organization Claim
-   * @param  index Index value of Claim.
+   * @param claimId UUID of the claim being approved
    * @param adminContractAddress Contract Address for Endaoment Admin
    */
-  function approveClaim(uint256 index, address adminContractAddress)
+  function approveClaim(string calldata claimId, address adminContractAddress)
     public
     onlyAdminOrRole(adminContractAddress, IEndaomentAdmin.Role.REVIEWER)
   {
-    Claim storage claim = claims[index];
-    emit ClaimApproved(claim);
-    orgWallet = claim.desiredWallet;
+    Claim storage claim = pendingClaims[claimId];
+    require(claim.desiredWallet != address(0), "Org: Cannot approve a claim with no wallet");
+
+    emit ClaimApproved(claimId, claim);
+
+    activeClaim = claim;
+    delete pendingClaims[claimId];
+  }
+
+  /**
+   * @notice Rejecting Organization Claim
+   * @param claimId UUID of the claim being rejected
+   * @param adminContractAddress Contract Address for Endaoment Admin
+   */
+  function rejectClaim(string calldata claimId, address adminContractAddress)
```

Comment:
require stmt to check fro claimId existence.

Reply:
What do you want to check, here? Just that it's not the empty string? I don't think that actually buys us much in terms of preventing harm.

Reply:
Add lines like:
```
  Claim storage claim = pendingClaims[claimId];
   require(claim.desiredWallet != address(0), "Org: Cannot reject a claim with no wallet");
```
I have a hunch the auditors will see this lacking a check and push it back to us.

Reply:
It's one thing to check if an address is 0. For claimId, though, we're expecting it to be a UUID string. Validating a properly formatted UUID string in Solidity would be painful, expensive, and (IMO) not add much value.

Reply:
No I just want a check that the address is not 0. I agree it is not necessary to look at the UUID, that is our deal to ensure it is the correct value.

Reply:
Here's my proposed solution: https://github.com/endaoment/endaoment-contracts/pull/75/files#r466737032

If we verify the `claimId` is not empty when it's first created, that way you have no need to check it here

Reply:
We ended up checking that `claimId` is not empty in both the create step and reject step